open_prices/proofs/ml.py

```diff
@@ -654,15 +690,63 @@ def run_and_save_proof_type_prediction(
     )
 
 
+def run_and_save_receipt_extraction_prediction(
+    image: Image, proof: Proof, overwrite: bool = False
+) -> ProofPrediction | None:
+    """Run the receipt extraction model and save the prediction in
+    ProofPrediction table.
+
+    :param image: the image to run the model on
+    :param proof: the Proof instance to associate the ProofPrediction with
+    :param overwrite: whether to overwrite existing prediction, defaults to
+        False
+    :return: the ProofPrediction instance created, or None if the prediction
+        already exists and overwrite is False
+    """
+    if proof.type != proof_constants.TYPE_RECEIPT:
```

Comment:
I added a commit to move this check here (similar to `run_and_save_price_tag_detection`)